src/coreclr/scripts/superpmi_benchmarks.py

```diff
@@ -209,36 +209,48 @@ def build_and_run(coreclr_args, output_mch_name):
-                            " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
-                            "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
+                             f"--partition-index {partition_index} --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput " \
+                             f"--envVars DOTNET_JitName:{shim_name} DOTNET_ZapDisable:1 DOTNET_ReadyToRun:0"
+
     elif benchmark_binary.lower().startswith("demobenchmarks"):
         # Disable ReadyToRun so we always JIT R2R methods and collect them
         collection_command = f"{dotnet_exe} {benchmarks_dll}  -f *CollisionBatcherTaskBenchmarks.* *GroupedCollisionTesterBenchmarks.* *GatherScatterBenchmarks.* " \
-                            " *OneBodyConstraintBenchmarks.* *TwoBodyConstraintBenchmarks.* *ThreeBodyConstraintBenchmarks.* *FourBodyConstraintBenchmarks.* " \
-                            " *SweepBenchmarks.* *ShapeRayBenchmarks.* *ShapePileBenchmark.* *RagdollTubeBenchmark.* " \
-                            f" --corerun {os.path.join(core_root, corerun_exe)} --envVars DOTNET_JitName:{shim_name} " \
-                            " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
-                            "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
+                              " *OneBodyConstraintBenchmarks.* *TwoBodyConstraintBenchmarks.* *ThreeBodyConstraintBenchmarks.* *FourBodyConstraintBenchmarks.* " \
+                              " *SweepBenchmarks.* *ShapeRayBenchmarks.* *ShapePileBenchmark.* *RagdollTubeBenchmark.* " \
+                             f" --corerun {os.path.join(core_root, corerun_exe)} " \
+                               "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput " \
+                             f" --envVars DOTNET_JitName:{shim_name} DOTNET_ZapDisable:1 DOTNET_ReadyToRun:0"
     else:
         # Disable ReadyToRun so we always JIT R2R methods and collect them
-        collection_command = f"{dotnet_exe} {benchmarks_dll}  --filter \"*\" --corerun {os.path.join(core_root, corerun_exe)} --envVars DOTNET_JitName:{shim_name} " \
-                            " DOTNET_ZapDisable:1  DOTNET_ReadyToRun:0 " \
-                            "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput"
+        collection_command = f"{dotnet_exe} {benchmarks_dll}  --filter \"*\" --corerun {os.path.join(core_root, corerun_exe)} " \
+                              "--iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart --logBuildOutput " \
+                             f"--envVars DOTNET_JitName:{shim_name} DOTNET_ZapDisable:1 DOTNET_ReadyToRun:0"
+
+    if coreclr_args.tiered_pgo:
```

Comment:
do you mind also pulling the ` --iterationCount 1 --warmupCount 0 --invocationCount 1 --unrollFactor 1 --strategy ColdStart ` in common case?

Also, shouldn't you be passing `iterationCount`, etc. if for tiered/pgo runs? And a note about `DOTNET_ZapDisable:1 DOTNET_ReadyToRun:0` too. Do we still want them for tiered/pgo runs?

Reply:
There is no common case right now, are you saying I should factor the common aspects out (or at least the common trailing elements)?

I wasn't planning on changing how we collect at this point. I would have to study the impact of altering BDN strategy to see if it adds value.

Reply:
> There is no common case right now, are you saying I should factor the common aspects out (or at least the common trailing elements)?

Yes, that's what I meant.

> I wasn't planning on changing how we collect at this point. I would have to study the impact of altering BDN strategy to see if it adds value.

Sure, we can limit this PR for just the changes you have currently and rethink about longer term strategy of how we should collect them in a later PR.